applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm

```diff
@@ -67,6 +67,18 @@
 		{
 			setTimeout(function(){ output.innerHTML=""}, timeout);
 		}
+	}
+		var vpn_file = document.getElementById("ovpn_file").value;
+
+		if(!vpn_name && !vpn_name.length) {
+			let vpn_file_name = vpn_file.replace('C:\\fakepath\\','').replace('.ovpn','').replace(/[^\x00-\x7F]|[\s\.!@#$%^&*()\-+=\[\]{};':"\\|,<>\/?]/g,'');
```

Comment:
I find the regex difficult to understand. Isn't there something simple here?
Maybe a path split see https://www.geeksforgeeks.org/how-to-get-the-file-name-from-full-path-using-javascript/

I also think that the assumption that it is about **windows** is not correct either.

Reply:
The regex was already in place and it checks the name for illegal characters. You can check the code where the name is setup manually.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/FileList

Then avoid all the regex and just get the pure file name.